s3transfer/manager.py

```diff
@@ -497,23 +501,29 @@ def shutdown(self, cancel=False):
         :param cancel: If True, calls TransferFuture.cancel() for
             all in-progress in transfers. This is useful if you want the
             shutdown to happen quicker.
+
+        :type cancel_msg: str
+        :param cancel_msg: The message to specify if canceling all in-progress
+            transfers.
         """
         if cancel:
             # Cancel all in-flight transfers if requested, before waiting
             # for them to complete.
-            self._coordinator_controller.cancel()
+            self._coordinator_controller.cancel(cancel_msg)
         try:
             # Wait until there are no more in-progress transfers. This is
             # wrapped in a try statement because this can be interrupted
             # with a KeyboardInterrupt that needs to be caught.
             self._coordinator_controller.wait()
-        finally:
+        except KeyboardInterrupt as e:
             # If not errors were raised in the try block, the cancel should
             # have no coordinators it needs to run cancel on. If there was
             # an error raised in the try statement we want to cancel all of
             # the inflight transfers before shutting down to speed that
             # process up.
-            self._coordinator_controller.cancel()
+            self._coordinator_controller.cancel('keyboard interrupt')
```

Comment:
I am not sure of a great way of testing that the message has ``'keyboard interrupt'``. What I need to do is be able to inject a KeyboardInterrupt straight into this method but there is not really anything for me to hook into because the coordinator controller is not exposed in the transfer manager and I really do not want to expose this. Does anyone have ideas on this?